File: src/main/java/kr/mjc/rachel/jdbc/DataSourceFactory.java
package kr.mjc.rachel.jdbc;

import org.mariadb.jdbc.MariaDbDataSource;

import javax.sql.DataSource;
import java.io.IOException;
import java.sql.SQLException;
import java.util.Properties;

public class DataSourceFactory {

  private static final DataSource dataSource = createDataSource();

  private static DataSource createDataSource() {
    Properties props = new Properties();
    try {
      props.load(DataSourceFactory.class.getResourceAsStream(
          "/application.properties"));
      return new MariaDbDataSource(props.getProperty("db.url"));
    } catch (IOException | SQLException e) {
      throw new RuntimeException(e);
    }
  }

  public static DataSource getDataSource() {
    return dataSource;
  }
}

File: src/main/kotlin/kr/mjc/jacob/jdbc/DataSourceFactory.kt
package kr.mjc.jacob.jdbc

import org.mariadb.jdbc.MariaDbDataSource
import java.util.*
import javax.sql.DataSource

class DataSourceFactory {
  companion object {
    private val props = Properties().apply {
      load(DataSourceFactory::class.java.getResourceAsStream(
          "/application.properties"))
    }

    val dataSource: DataSource = MariaDbDataSource(props.getProperty("db.url"))
  }
}
